cross-edits-detection/cross-edit-suspicious-patterns.ts

```diff
@@ -0,0 +1,438 @@
+  }
+
+  // Used for both "author" mode and "article" mode
+  async getScoreAndProcess(){
+    var title = this.title;
+    var author = this.author;
+    var editsList = this.editsList;
+
+    var scores = new Array(Math.min(this.windowSize, editsList.length));
+    var previousRevisionInfos = new Array(Math.min(this.windowSize, editsList.length));
+    for (var i = 0; i < scores.length; i++) {
+      //Only take ORES_DAMAGING score 
+      //If ORES Scores are missing, skip this edit entirely. 
+      if(editsList[i].oresscores.damaging == undefined) {
+        var missingScoreString = "";
+        missingScoreString += "Title: " + title + " Author: " + author + "\n";
+        missingScoreString += "ORES Scores are missing. Hence no detection is performed. \n";
+        missingScoreString += "Timestamp: " + editsList[0].timestamp + "\n";
+        console.log(missingScoreString);
+        var decisionInfo = {
+          mode: this.mode,
+          type: this.type,
+          author: this.author,
+          recipient: this.recipient, 
+          percentage: (this.avg * 100).toFixed(0), 
+          previousRevisionInfos: this.previousRevisionInfos,
+        }
+        return decisionInfo;
+      }
+      scores[i] = editsList[i].oresscores.damaging.true;
+      if (this.mode == "author") {
+        var editInfo = {
+          author: this.author,
+          title: String(editsList[i].title),
+          score: (scores[i] * 100).toFixed(0),
+          timestamp: editsList[i].timestamp,
+          parentid: editsList[i].parentid,
+        };
+      } else if (this.mode == "article") {
+        var editInfo = {
+          author: String(editsList[i].user),
+          title: this.title,
+          score: (scores[i] * 100).toFixed(0),
+          timestamp: editsList[i].timestamp,
+          parentid: editsList[i].parentid,
+        };
+      }
+      previousRevisionInfos[i] = editInfo;
+    }
+
+    var windowStart: number = editsList[editsList.length-1].timestamp;
+    var windowEnd: number = editsList[0].timestamp;
+    var windowStartDate: Date = new Date(windowStart);
+    var windowEndDate: Date = new Date(windowEnd);
+    var avg: number = scores.reduce((acc, e) => acc + e, 0) / scores.length;
+    var diff: number = avg - this.baseline;
+    this.windowStart = windowStart;
+    this.windowEnd = windowEnd;
+    this.avg = avg;
+    this.diff = diff;
+    this.previousRevisionInfos = previousRevisionInfos;
+    
+    if(diff > this.margin) {
+      if (this.mode == "author") {
+        var warnings = await this.getPreviousWarningsAuthor(author, windowEnd);
+        if (warnings.length > this.warningThreshold) {
+          this.type = "block";
+          this.recipient = await this.getRecipientForBlock();
+        }else{
+          this.type = "warning";
+          this.recipient = author;
+        }
+        this.writeNewDecisionAuthor(author, title, this.type, windowEndDate, this.recipient, windowStartDate, avg);
```

Comment:
Why a "getScoreAndProcess` will create an update to the `DecisionLog`? Can you separate what's writing and what is readonly?